src/lib/THERMAL/devThermal.cpp

```diff
@@ -110,11 +172,63 @@ static void timeoutFan()
         }
         else
         {
-            digitalWrite(GPIO_PIN_FAN_EN, HIGH);
+            if (GPIO_PIN_FAN_EN != UNDEF_PIN)
+            {
+                digitalWrite(GPIO_PIN_FAN_EN, HIGH);
+            }
+#if defined(PLATFORM_ESP32)
+            else if (GPIO_PIN_FAN_PWM != UNDEF_PIN)
+            {
+                ledcWrite(fanChannel, fanSpeeds[POWERMGNT::currPower()]);
+                DBGLN("Fan speed: %d (power) -> %d (pwm)", POWERMGNT::currPower(), fanSpeeds[POWERMGNT::currPower()]);
```

Comment:
Can you make this body into a function and call it from here and the identical code in the isOn block?